Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -3775,68 +3775,66 @@ internal async static Task ListenBucketNotificationsAsync_Test1(MinioClient mini
                 await Setup_Test(minio, bucketName);
 
                 var received = new List<MinioNotificationRaw>();
+                CancellationTokenSource source = new CancellationTokenSource();
+                CancellationToken token = source.Token;
 
                 List<EventType> eventsList = new List<EventType>();
                 eventsList.Add(EventType.ObjectCreatedAll);
                 ListenBucketNotificationsArgs listenArgs = new ListenBucketNotificationsArgs()
                                                                         .WithBucket(bucketName)
                                                                         .WithEvents(eventsList);
-                IObservable<MinioNotificationRaw> events = minio.ListenBucketNotificationsAsync(listenArgs);
+                IObservable<MinioNotificationRaw> events = minio.ListenBucketNotificationsAsync(listenArgs,token);
                 subscription = events.Subscribe(
                     ev => {
                         Console.WriteLine($"ListenBucketNotificationsAsync received: " + ev.json);
                         received.Add(ev);
+                        source.Cancel();
                     },
                     ex => Console.WriteLine("OnError: {0}", ex.Message),
                     () => Console.WriteLine($"ListenBucketNotificationsAsync finished")
                 );
                 await PutObject_Tester(minio, bucketName, objectName, null, contentType, 0, null, rsg.GenerateStreamFromSeed(1 * KB));
 
-                // wait for notifications
-                for (int attempt = 0; attempt < 10; attempt++) {
+                // validate received notifications
+                if (received.Count > 0) {
```

Comment:
If the `received.Count == 0`, we should not jump to the bottom and declare a `PASS`.

Reply:
I had accidentally removed a line in the test that handles this - pls look again.